### ISSUE TITLE
Add support for AreaLights

### ISSUE THREAD

**Issue by sebavan** [BabylonJS/Babylon.js#7866]
Based on those researches, integrate supports for Area Lights:
* https://eheitzresearch.wordpress.com/415-2/
* https://eheitzresearch.wordpress.com/757-2/

**Comment by undefinedlee** [comment]
Which version of this feature is expected to be implemented

**Comment by deltakosh** [comment]
We don't know yet. This is something we are looking for help if you are motivated

**Comment by Pryme8** [comment]
Is this still something we want? I was looking at the code it would not really be that hard to convert.

**Comment by deltakosh** [comment]
oh yes we want it. but we want it as a regular source (so for all meshes)

**Comment by Ivolutio** [comment]
Maybe PlayCanvas their implementation (light shapes) can also help. They have predefined shapes though. 
Docs: https://developer.playcanvas.com/en/user-manual/graphics/lighting/lights/#light-shapes
Example: https://playcanvas.github.io/#/graphics/area-lights

**Comment by lokiiarora** [comment]
Do we want shadow support as well?

**Comment by Popov72** [comment]
I don't think so, soft shadows can be simulated to some extent with PCSS or some other filter method. Generating accurate shadows for area lights would be quite time consuming I think.
